src/app_service/service/transaction/service.nim

```diff
@@ -209,6 +214,25 @@ QtObject:
     )
     self.threadpool.start(arg)
 
+  proc onFetchDecodedTxData*(self: Service, response: string) {.slot.} =
+    try:
+      let data = parseJson(response)
+      if data.len == 0 or not data.hasKey("result"):
+        return
+      self.events.emit(SIGNAL_TRANSACTION_DECODED, TransactionDecodedArgs(dataDecoded: $data["result"], txHash: data["txHash"].getStr))
```

Comment:
@Cuteivist if we do an aync action, you have to send a signal even the data are not received successfully, just in case if we're displaying any loading state on the UI state, we need to stop it in that case.

Reply:
Although right now there isn't any UI state to reset it's future proof idea to send signal if UI will want to handle any states.

Updated @saledjenic